### PR TITLE
implemented nullable safety

### PR DESCRIPTION
**Pull request by kechankrisna**

### CODE REVIEW

**Comment by rus0n** [comment]
how can i pull this code in the old version code? you know?

**Comment by vishnuagbly** [comment]
unfortunately, it looks like this project has been abandoned, @kechankrisna if you are interested would you like to create a new package out of your null safety version? I would also like to help as needed.

**Comment by JuniorEC** [comment]
Where is new repository?

**Comment by paurakhsharma** [comment]
If you are in urgent need you can add this in your `pubspec.yml`

```
flutter_bluetooth_serial:
    git:
      url: git://github.com/kechankrisna/flutter_bluetooth_serial
      ref: nullable
```

File: lib/FlutterBluetoothSerial.dart [hunk 1]
```diff
@@ -195,20 +194,21 @@ class FlutterBluetoothSerial {
 
   /// Returns list of bonded devices.
   Future<List<BluetoothDevice>> getBondedDevices() async {
-    final List list = await _methodChannel.invokeMethod('getBondedDevices');
+    final List list = await (_methodChannel.invokeMethod('getBondedDevices')
+        as FutureOr<List<dynamic>>);
```
Comment (DCrow) [inline lib/FlutterBluetoothSerial.dart:198]: In current flutter v 2.2.0 this fails with
```dart
type 'Future<dynamic>' is not a subtype of type 'FutureOr<List<dynamic>>' in type cast
I/flutter (21832): #0      FlutterBluetoothSerial.getBondedDevices
package:flutter_bluetooth_serial/FlutterBluetoothSerial.dart:198

```

**Comment by nightscape** [comment]
@kechankrisna I got contributor permissions from @edufolly.
If you could check that this works with Flutter 2.2.0, I'll get it merged  🎉

**Pull request merged**